hs_core/tasks.py

```diff
@@ -158,6 +168,77 @@ def nightly_periodic_task_check():
         file.write(timezone.now().strftime('%m/%d/%y %H:%M:%S'))
 
 
+@celery_app.task(ignore_result=True, base=HydroshareTask)
+def nightly_repair_resource_files():
+    """
+    Run repair_resource management command on resources updated in the last day
+    """
+    from hs_core.management.utils import check_time, repair_resource
+    start_time = time.time()
+    cuttoff_time = timezone.now() - timedelta(days=1)
+    recently_updated_resources = BaseResource.objects.filter(updated__gte=cuttoff_time)
```

Comment:
Should we be repairing published resources? It is more of a policy question. Do we know if repair resource works for a published resource?

Reply:
Good question. I thought of this and then promptly forgot :)
So yes, repair_resource does work for a published resource. I just tested it locally and actually something perhaps in [these optimizations](https://github.com/hydroshare/hydroshare/compare/develop...optimize-resource-repair-command#diff-e6fd5b9c81f5b092cfb9f7073b3d0155d9552adfaac9af6f35f7b6d1d2b56a40) causes `res_file` to potentially be unbound--when I ran it I got 
  

> File "/hydroshare/hs_core/management/commands/repair_resource.py", line 57, in handle
>     repair_resource(resource, logger)
>   File "/hydroshare/hs_core/management/utils.py", line 669, in repair_resource
>     _, ingest_count = ingest_irods_files(resource,
>   File "/hydroshare/hs_core/management/utils.py", line 489, in ingest_irods_files
>     return __ingest_irods_directory(resource,
>   File "/hydroshare/hs_core/management/utils.py", line 548, in __ingest_irods_directory
>     file_id=res_file.pk, fail_feedback=False)
> UnboundLocalError: local variable 'res_file' referenced before assignment

but anyway, I don't think that's because it was run on a published resource. When I run repair_resource again, it finishes successfully and repairs even a published resource.

So you're right, we need to make a policy decision. My guess is that we want to filter out published resources.

Reply:
Yeah it was a bug there before. I can fix it or you can fix it by updating this line:
https://github.com/hydroshare/hydroshare/blob/771f81e766a83d8a18b7e7adbece7d70e9436292/hs_core/management/utils.py#L543
to
`res_file = link_irods_file_to_django(resource, fullpath)`

Reply:
added. thanks